lib/dom_html.ml

```diff
@@ -461,8 +461,8 @@ module Prop = struct
 
   (* react-specific *)
 
-  let dangerouslySetInnerHTML ~(__html : string) =
-    any "dangerouslySetInnerHTML" (Dom.makeInnerHtml ~__html)
+  let dangerouslySetInnerHTML (safe_html : Dom.DangerouslySetInnerHTML.t) =
+    any "dangerouslySetInnerHTML" safe_html
```

Comment:
What's the reason for this change? Just uniformity?

Reply:
Uniformity for the sake of making the PPX simpler I assume?

Reply:
Yes, exactly.

That, and also, I thought that it would makes sense to force consumers to call `React.Dom.makeHtml` explicitly, to emulate what React.js does with the `{__html}` object: https://zhenyong.github.io/react/tips/dangerously-set-inner-html.html. The idea as I understand from the link above is to avoid situations where some string is passed directly.

Reply:
From an OCaml perspective, I think this is actually quite misleading, because you'd expect a construct like this to actually do something, like maintain some invariant. And I'd rather use something that actually does, and communicates that more clearly, like `SafeString.t`.

The problem is that the concept of "safe" here is extremely vague. It's a string containing only the "safe" set of HTML tags, where that set depends entirely on what's expected in a given context. It's difficult enough to create a HTML sanitizer, even more so to create a well-typed one that can sanitize selectively. We could start by making the type though, something like ``[ `a | `em | `strong | `p | `br ] Sanitized.t``, and for now just have a `val make_unchecked : string -> _ t` to create it, effectively asserting that it's already been sanitized.

Reply:
I like this idea. I implemented it in https://github.com/ml-in-barcelona/jsoo-react/pull/128/commits/7767fc1a62f0530ea2e62ca50ff43672f50e522a. For now I didn't add any params to `SafeString.t` as I think it'd be more confusing while there are no checks being done. I added versions of `make_unchecked` with camel case and snake case, as I'd like to have all APIs supporting both style in the future 😅 
lmk what you think.